tasks/go.py

```diff
@@ -278,17 +292,24 @@ def deps(ctx, verbose=False):
     # We won't need this if/when we change to non-vendored modules
     ctx.run('go run github.com/goware/modvendor -copy="**/*.c **/*.h **/*.proto **/*.java"{}'.format(verbosity))
 
-    # delete gopsutil on windows, it get vendored because it's necessary on other platforms
-    if sys.platform == 'win32':
-        print("Removing PSUTIL on Windows")
-        ctx.run("rd /s/q vendor\\github.com\\shirou\\gopsutil")
+    # If github.com/DataDog/datadog-agent gets vendored too - nuke it
+    #
+    # This may happen as a result of having to introduce DEPPROJECTROOT
+    # in our builders to get around a known-issue with go dep, and the
+    # strange GOPATH situation in our builders.
+    #
+    # This is only a workaround, we should eliminate the need to resort
+    # to DEPPROJECTROOT.
+    if os.path.exists('vendor/github.com/DataDog/datadog-agent'):
+        print("Removing vendored github.com/DataDog/datadog-agent")
+        shutil.rmtree('vendor/github.com/DataDog/datadog-agent')
```

Comment:
Are we sure we want to use relative path for this ?

Reply:
This code was present before (removed in #7174) and I am adding it back to prevent vendoring nested modules so it shouldn't be a problem. The comment is outdated though, let me update that.